### PR TITLE
Migrate serving support

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
I didn't update all of the hosting-related integ tests because of the lack of custom `transform_fn` support in SageMaker Containers (which is my next task). (Also more integ test changes are coming anyway.)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker_mxnet_container/serving.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+
+def default_model_fn(model_dir, preferred_batch_size=None):
+    """Function responsible for loading the model. For more information, see
+    https://github.com/aws/sagemaker-python-sdk#model-loading.
+
+    Args:
+        model_dir (str): The directory where model files are stored.
+
```
Comment (ChoiByungWook) [inline src/sagemaker_mxnet_container/serving.py]: add doc for preferred_batch_size?

File: src/sagemaker_mxnet_container/serving.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+DEFAULT_MODEL_NAME = 'model'
+DEFAULT_MODEL_FILENAMES = {
+    'symbol': 'model-symbol.json',
+    'params': 'model-0000.params',
+    'shapes': 'model-shapes.json',
+}
+
+
+def default_model_fn(model_dir, preferred_batch_size=None):
```
Comment (ChoiByungWook) [inline src/sagemaker_mxnet_container/serving.py]: Does this value get populated with something other than None? I think it was 1 originally or takes the value from the environment class, but that was passed in directly originally.

https://github.com/aws/sagemaker-mxnet-container/blob/master/src/mxnet_container/serve/transformer.py#L110

Comment (icywang86rui) [inline src/sagemaker_mxnet_container/serving.py]: Does this only work with the model saved with the default save? Let's document that in the docstring.

File: src/sagemaker_mxnet_container/serving.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    Args:
+        model_dir (str): The directory where model files are stored.
+
+    Returns:
+        mxnet.mod.Module: the loaded model.
+    """
+    for f in DEFAULT_MODEL_FILENAMES.values():
+        path = os.path.join(model_dir, f)
+        if not os.path.exists(path):
+            raise ValueError('missing %s file' % f)
```
Comment (icywang86rui) [inline src/sagemaker_mxnet_container/serving.py]: I would like a bit more context with the error message - something like:

failed to load model with default_model_fn: missing %s file....

File: test/integ/test_gluon_hosting.py [hunk 4]
```diff
@@ -13,25 +13,26 @@
...
+from sagemaker.mxnet.model import MXNetModel
 
-import docker_utils
-import utils
+import local_mode
 
 
 # The image should support serving Gluon-created models.
-@pytest.mark.skip
 def test_gluon_hosting(docker_image, opt_ml, processor):
```
Comment (icywang86rui) [inline test/integ/test_gluon_hosting.py]: Does the mnist test use gluon? If not should we add a training test for gluon?

File: src/sagemaker_mxnet_container/serving.py (hunk 3 above)
Comment (laurenyu) [reply src/sagemaker_mxnet_container/serving.py]: done

File: test/integ/test_gluon_hosting.py (hunk 4 above)
Comment (laurenyu) [reply test/integ/test_gluon_hosting.py]: it doesn't. adding a training test with gluon is probably a good idea - I've noted it in the things I have left to do, and will address it in a later PR (time permitting)

File: setup.py [hunk 5]
```diff
@@ -59,6 +59,6 @@ def read(fname):
     install_requires=['sagemaker-containers==2.1', 'retrying==1.3.3'],
     extras_require={
         'test': ['tox', 'flake8', 'pytest', 'pytest-cov', 'pytest-xdist', 'mock', 'sagemaker',
-                 'requests==2.18.4', 'docker-compose']
+                 'requests==2.18.4', 'docker-compose', 'mxnet']
```
Comment (icywang86rui) [inline setup.py]: does this get installed when we pip install sagemaker-mxnet-container? Is going to interfere with the mxnet binary we installed directly in the dockerfile?

Comment (laurenyu) [reply setup.py]: only gets installed for tests, though yeah, freezing the version is probably a good idea

File: setup.py [hunk 6]
```diff
@@ -59,6 +59,6 @@ def read(fname):
     install_requires=['sagemaker-containers==2.1', 'retrying==1.3.3'],
```
Comment (ChoiByungWook) [inline setup.py]: https://github.com/aws/sagemaker-pytorch-container/blob/master/setup.py#L38

Your unit tests might fail?

Comment (laurenyu) [reply setup.py]: updated sagemaker-containers version (I assume that's what you're referring to)

**Pull request merged**